backend/src/api/middlewares/rate-limiters.ts

```
@@ -166,3 +167,258 @@ export const sendEmailOTPLimiter = [
+ * Public URL of the live override file. Expected shape:
+ * `{ "functions"?: number, "deployments"?: number, "compute"?: number }`.
+ * Any missing or invalid (non-positive-integer) field falls back to the default.
+ *
+ * Self-hosters can pin their own config by setting
+ * INSFORGE_WRITE_RATE_LIMIT_CONFIG_URL. Plain HTTPS is used instead of the
+ * AWS SDK so the fetch works on instances without AWS credentials and never
+ * gets rejected because the runtime's signing identity lacks read access to
+ * a public bucket.
+ */
+const DEFAULT_WRITE_ENDPOINT_LIMITS_URL =
+  'https://config.insforge.dev/resource-rate-limits.json';
```

ESLint

[error] Delete `⏎·`

(prettier/prettier)